tests/pytorch/r2.0/common.libsonnet

```diff
@@ -0,0 +1,138 @@
+  },
+  PyTorchTest:: common.PyTorchTest + r2_0 {
+    local config = self,
+
+    podTemplate+:: {
+      spec+: {
+        initContainerMap+:: {
+          'tpu-version': {
+            image: config.podTemplate.spec.containerMap.train.image,
+            env+: [
+              {
+                name: 'TPU_NAME',
+                valueFrom: {
+                  fieldRef: {
+                    fieldPath: "metadata.annotations['name.cloud-tpus.google.com/train']",
+                  },
+                },
+              },
+            ],
+            command: [
+              'python3',
+              '-c',
+              |||
+                import importlib_metadata
+                import os
+                import re
+
+                import cloud_tpu_client
+
+                requirements = importlib_metadata.requires('torch_xla')
+                libtpu_pattern = r'libtpu-nightly ?@ https:\/\/storage.googleapis.com\/cloud-tpu-tpuvm-artifacts\/wheels\/libtpu-nightly\/libtpu_nightly-\d.\d.dev(\d{8})-\w+-\w+-\w+.whl'
+                libtpu_matches = [
+                  re.findall(libtpu_pattern, req)[0]
+                  for req in requirements
+                  if re.match(libtpu_pattern, req)
+                ]
+                assert len(libtpu_matches) == 1, f'{len(libtpu_matches)} matches in {requirements} (pattern: `{libtpu_pattern}`)'
+                libtpu_date = libtpu_matches[0]
+                print('libtpu date:', libtpu_date)
+
+                ctc = cloud_tpu_client.Client(tpu=os.path.basename('$(TPU_NAME)'), zone=os.path.dirname('$(TPU_NAME)'))
+                ctc.wait_for_healthy()
+                ctc.configure_tpu_version(f'pytorch-nightly-dev{libtpu_date}', restart_type='always')
+                ctc.wait_for_healthy()
+              |||,
+            ],
+          },
+        },
+      },
+    },
+  },
+  PyTorchXlaDistPodTest:: common.PyTorchXlaDistPodTest + r2_0,
+  PyTorchGkePodTest:: common.PyTorchGkePodTest + r2_0,
+  Functional:: mixins.Functional {
+    schedule: null,
+    tpuSettings+: {
+      preemptible: false,
+    },
+  },
+  Convergence:: mixins.Convergence,
+  PyTorchTpuVmMixin:: experimental.PyTorchTpuVmMixin {
+    local config = self,
+
+    tpuSettings+: {
+      softwareVersion: if config.accelerator.version < 4 then
+        'tpu-vm-pt-2.0'
+      else
+        'tpu-vm-v4-pt-2.0',
+      tpuVmPytorchSetup: |||
+        # No need to check out the PyTorch repository, but check out PT/XLA at
+        # pytorch/xla anyway
+        mkdir pytorch
+        cd pytorch
+        git clone https://github.com/pytorch/xla.git -b r2.0
+      |||,
+    },
+    podTemplate+:: {
+      spec+: {
+        initContainerMap+:: {
+          'tpu-version': null,
+        },
+      },
+    },
+  },
+  datasetsVolume: volumes.PersistentVolumeSpec {
+    name: 'pytorch-datasets-claim',
+    mountPath: '/datasets',
+  },
+  GpuMixin:: {
+    local config = self,
+    imageTag+: '_cuda_11.2',
```

Comment:
Which cuda image(s) should we test? 11.7 and 11.8 both? @JackCaoG

Reply:
let's do 11.8 and see if it works.